iceoryx2-services/discovery/src/service_discovery/service.rs

```diff
@@ -0,0 +1,331 @@
+                                      // case IMO
+    pub fn spin(&mut self) -> Result<(Vec<&ServiceDetails<S>>, Vec<ServiceDetails<S>>), SpinError> {
+        // Detect changes
+        let (added_ids, removed_services) = self.tracker.sync(&self.iceoryx_config);
+        let changes_detected = !added_ids.is_empty() || !removed_services.is_empty();
+
+        // Publish
+        let mut added_services = Vec::new();
+        for id in &added_ids {
+            if let Some(service) = self.tracker.get(id) {
+                if !self.discovery_config.include_internal
+                    && is_internal_service(service.static_details.name())
+                {
+                    continue;
+                }
+                self.publish(&Discovery::Added(service.static_details.clone()))
+                    .unwrap();
+
+                // Keep track of added services to be returned.
+                added_services.push(service);
+            }
+        }
+
+        for service in &removed_services {
+            if !self.discovery_config.include_internal
+                && is_internal_service(service.static_details.name())
+            {
+                continue;
+            }
+            self.publish(&Discovery::Removed(service.static_details.clone()))
+                .unwrap();
+        }
+
+        // Notify
+        if let Some(notifier) = &mut self.notifier {
+            if changes_detected {
+                notifier.notify().map_err(|_| SpinError::NotifyFailure)?;
+            }
+        }
+
+        Ok((added_services, removed_services))
+    }
+
+    fn publish(&self, discovery: &Discovery) -> Result<(), SpinError> {
+        if let Some(publisher) = &self.publisher {
+            // This intermediate struct is inefficient ... need to find a serialization
+            // solution that can serialize directly to the loaned buffer.
+            let serialized =
+                serde_json::to_vec(discovery).map_err(|_| SpinError::PublishFailure)?;
+
+            let sample = publisher
+                .loan_slice_uninit(serialized.len())
+                .map_err(|_| SpinError::PublishFailure)?;
+
+            let sample = sample.write_from_slice(serialized.as_slice());
+            sample.send().map_err(|_| SpinError::PublishFailure)?;
+        }
+        Ok(())
+    }
+}
+
+/// Returns the service name used by the service discovery service.
+///
+/// This function returns a reference to a lazily initialized static `ServiceName`
+/// instance. The service name is constructed by concatenating the internal service
+/// prefix with the discovery service name.
+///
+/// # Returns
+///
+/// A reference to the static `ServiceName` instance used for the discovery service.
+///
+/// # Panics
+///
+/// This function will panic during the first call if the service name is invalid,
+/// which should never happen with the predefined constants.
+///
+/// # Examples
+///
+pub fn service_name() -> &'static ServiceName {
```

Comment:
Wouldn't it make sense to add this to the config as well?

Reply:
I like the idea of fixed well-known service names. Ping @elBoberido what is your opinion?

Reply:
I fully support well known service names. I would even go so far and reserve a namespace for iceoryx, e.g. a user cannot use `"/iox2/blup"` as service name. But that might be not that easy to achieve. Nevertheless, a blessed namespace for iceoryx2 services would make sense.